test/e2e/tests/test_references.py

```diff
@@ -0,0 +1,163 @@
+def referred_policy_name():
+    return random_suffix_name("referred-policy", 24)
+
+
+@pytest.fixture(scope="module")
+def referring_role(referred_policy_name):
+    role_name = random_suffix_name("referring-role", 24)
+    role_desc = "a role that refers to a policy"
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements['ROLE_NAME'] = role_name
+    replacements['ROLE_DESCRIPTION'] = role_desc
+    replacements['MAX_SESSION_DURATION'] = str(MAX_SESS_DURATION)
+    replacements['POLICY_NAME'] = referred_policy_name
+
+    resource_data = load_resource(
+        "role_referring",
+        additional_replacements=replacements,
+    )
+
+    ref = k8s.CustomResourceReference(
+        CRD_GROUP, CRD_VERSION, ROLE_RESOURCE_PLURAL,
+        role_name, namespace="default",
+    )
+    k8s.create_custom_resource(ref, resource_data)
+    cr = k8s.wait_resource_consumed_by_controller(ref)
+
+    assert cr is not None
+    assert k8s.get_resource_exists(ref)
+
+    # NOTE(jaypipes): We specifically do NOT wait for the Role to exist in
+    # the IAM API here because we will create the referred-to Policy and
+    # wait for the reference to be resolved
+
+    yield (ref, cr, role_name)
+
+    if k8s.get_resource_exists(ref):
+        # If all goes properly, we should not hit this because the test cleans
+        # up the child resource before exiting...
+        _, deleted = k8s.delete_custom_resource(
+            ref,
+            period_length=DELETE_ROLE_TIMEOUT_SECONDS,
+        )
+        assert deleted
+
+        role.wait_until_deleted(role_name)
+
+
+@pytest.fixture(scope="module")
+def referred_policy(referred_policy_name):
+    policy_desc = "a referred-to policy"
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements['POLICY_NAME'] = referred_policy_name
+    replacements['POLICY_DESCRIPTION'] = policy_desc
+
+    resource_data = load_resource(
+        "policy_simple",
+        additional_replacements=replacements,
+    )
+
+    ref = k8s.CustomResourceReference(
+        CRD_GROUP, CRD_VERSION, POLICY_RESOURCE_PLURAL,
+        referred_policy_name, namespace="default",
+    )
+    k8s.create_custom_resource(ref, resource_data)
+    cr = k8s.wait_resource_consumed_by_controller(ref)
+
+    cr = k8s.get_resource(ref)
+    assert cr is not None
+    assert 'status' in cr
+    assert 'ackResourceMetadata' in cr['status']
+    assert 'arn' in cr['status']['ackResourceMetadata']
+    policy_arn = cr['status']['ackResourceMetadata']['arn']
+
+    policy.wait_until_exists(policy_arn)
+
+    assert cr is not None
+    assert k8s.get_resource_exists(ref)
+
+    yield (ref, cr, policy_arn)
+
+    _, deleted = k8s.delete_custom_resource(
+        ref,
+        period_length=DELETE_POLICY_TIMEOUT_SECONDS,
+    )
+    assert deleted
+
+    policy.wait_until_deleted(policy_arn)
+
+
+@service_marker
+@pytest.mark.canary
+class TestReferences:
+    def test_references(self, referring_role, referred_policy):
```

Comment:
Rename to denote as the `role-policy` reference relationship?
```suggestion
def test_policy_role_reference(self, referring_role, referred_policy):
```

Reply:
done.